### PR TITLE
production/GFS.v16: update module initialization script

### PR DESCRIPTION
**Pull request by junwang-noaa**
Orion platform is added to module_setup script to allow that module files are properly purged on Orion.

### CODE REVIEW

**Comment by MinsukJi-NOAA** [comment]
When I made the similar changes for the S2S model, the path was /apps/lmod/init/$__ms_shell.
I just checked on Orion, and it looks like this has been updated on Jun 22 to /apps/lmod/lmod/init/$__ms_shell.

**Comment by junwang-noaa** [comment]
Minsuk,

This change should be made in NEMS, I am curious how you make it in s2s
model. I see the changes were made in NEMS develop branch. This PR is for
NEMS production/GFS.v16 branch, which is used for GFSv16 implementation.

**Comment by MinsukJi-NOAA** [comment]
Sorry, I meant the NEMS develop branch. Can we double check on the path on Orion?

File: src/conf/module-setup.csh.inc [hunk 1]
```diff
@@ -14,6 +14,12 @@ else if ( { test -d /scratch1 -a ! -d /scratch } ) then
         source /apps/lmod/lmod/init/$__ms_shell
     endif
     module purge
+else if ( { test -d /work/noaa } ) then
+    # We are on Orion
+    if ( ! { module help >& /dev/null } ) then
+      source /apps/lmod/init/$__ms_shell
```
Comment (MinsukJi-NOAA) [inline src/conf/module-setup.csh.inc]: should this be `/apps/lmod/lmod/init/$__ms_shell`

File: src/conf/module-setup.sh.inc [hunk 2]
```diff
@@ -28,6 +28,12 @@ elif [[ -d /scratch1 && ! -d /scratch ]] ; then
         source /apps/lmod/lmod/init/$__ms_shell
     fi
     module purge
+elif [[ -d /work/noaa ]] ; then
+    # We are on Orion
+    if ( ! eval module help > /dev/null 2>&1 ) ; then
+        source /apps/lmod/init/$__ms_shell
```
Comment (MinsukJi-NOAA) [inline src/conf/module-setup.sh.inc]: Similarly here, `/apps/lmod/lmod/init`

**Comment by junwang-noaa** [comment]
That's weird, I remember I checked the path when I made this update, I
first tried hera's path, which is /apps/lmod/lmod/init, but that path  does
not exist,  so I changed /apps/lmod/init. Now I do see
/apps/lmod/lmod/init  exist, but /apps/lmod/init does not exist.  Will
change back.

**Comment by MinsukJi-NOAA** [comment]
Yeah, I can confirm that it was /apps/lmod/init :)

**Comment by junwang-noaa** [comment]
They must be changed during Orion maintenance, the directory is changed on
6/22

Orion-login-2[129] junwang$ ls -l /apps/lmod
total 0
drwxr-xr-x 11 root root 176 Jun 22 15:35 8.3.17
lrwxrwxrwx  1 root root   6 Jun 22 15:35 lmod -> 8.3.17

**Review by MinsukJi-NOAA — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**